tests/tests_pytorch/tuner/test_scale_batch_size.py

```diff
@@ -317,7 +317,7 @@ def test_dataloader_reset_with_scale_batch_size(tmp_path, caplog, scale_method,
     assert caplog.text.count("greater or equal than the length") == int(new_batch_size == dataset_len)
 
     assert trainer.train_dataloader.batch_size == new_batch_size
-    assert trainer.val_dataloaders.batch_size == init_batch_size
+    assert trainer.val_dataloaders.batch_size == new_batch_size
```

Comment:
Please note that this line is updated for the following reason:

Current `_scale_batch_size` instantiates the `_FitLoop` with `trainer.limit_val_batches = steps_per_trial`. This results in `_EvaluationLoop._max_batches` being initialized in a [nontrivial way withing the setup_data method](https://github.com/Lightning-AI/lightning/blob/1210deb5a0ff720b0ad66e38149cd28471d760ca/src/lightning/pytorch/loops/evaluation_loop.py#L194). As such we are in the following situation:
- current `_EvaluationLoop` (within the `_FitLoop`) initialized with `_max_batches` non suitable for an actual train
- there are no easy way to reset `_max_batches` without calling to `_EvaluationLoop.setup_data`  
- calling to `_EvaluationLoop.setup_data` would change batch size to the current size that was found by the `_scale_batch_size`


Imo this is pretty reasonable to change `assert trainer.val_dataloaders.batch_size` to `new_batch_size`, as this solves aforementioned issue, and there are no guarantees that `init_batch_size` were even possible to fit into the memory in the first place